chore(sdk): Update passkeys documentation (#649)
safe-global/safe-docs: pages/sdk/signers/passkeys.mdx

  {/* <!-- vale on --> */}

>   After generating the `passkeyCredential` object, you need to create the signer. This signer will be an `PasskeyArgType` object containing the `rawId` and the `coordinates` information.

Reviewer: ```suggestion
  After generating the `passkeyCredential` object, you need to create the signer. This signer will be a `PasskeyArgType` object containing the `rawId` and the `coordinates` information.
```

    throw Error('Passkey creation failed: No credential was returned.')
  }

  const passkeySigner = await Safe.createPasskeySigner(passkeyCredential)
  ```

  {/* <!-- vale on --> */}

  At this point, it's critical to securely store the information in the `passkeySigner` object in a persistent service. Losing access to this data will result in the user being unable to access their passkey and, therefore, their Safe Smart Account.

  ### Get the provider and signer

  Once the passkey is created, you need the `provider` and `signer` properties required to instantiate the Safe\{Core\} SDK kits.

> Check [how to initialize the Protocol Kit](../../reference-sdk-protocol-kit/initialization/init.mdx)

Reviewer: ```suggestion
  Check [how to initialize the Protocol Kit](../../reference-sdk-protocol-kit/initialization/init.mdx)
```

    throw Error('Passkey creation failed: No credential was returned.')
  }

  const passkeySigner = await Safe.createPasskeySigner(passkeyCredential)
  ```

  {/* <!-- vale on --> */}

  At this point, it's critical to securely store the information in the `passkeySigner` object in a persistent service. Losing access to this data will result in the user being unable to access their passkey and, therefore, their Safe Smart Account.

  ### Get the provider and signer

  Once the passkey is created, you need the `provider` and `signer` properties required to instantiate the Safe\{Core\} SDK kits.

>   [How to initialize the Protocol Kit](https://docs.safe.global/reference-sdk-protocol-kit/initialization/init)

Reviewer: ```suggestion
Check [how to initialize the Protocol Kit](../../reference-sdk-protocol-kit/initialization/init.mdx)
```

>   You can instantiate the provider using `viem` and the following imports:

Reviewer: Can you add the <Tabs/> component here like before?

  ```typescript
  import { createWalletClient, http } from 'viem'

>   import { sepolia } from 'viem/chains
>   const provider = createWalletClient({

Reviewer: ```suggestion
  import { sepolia } from 'viem/chains
  
  const provider = createWalletClient({
```

    throw Error('Passkey creation failed: No credential was returned.')
  }

  const passkeySigner = await Safe.createPasskeySigner(passkeyCredential)
  ```

  {/* <!-- vale on --> */}

  At this point, it's critical to securely store the information in the `passkeySigner` object in a persistent service. Losing access to this data will result in the user being unable to access their passkey and, therefore, their Safe Smart Account.

  ### Get the provider and signer

>   Once the passkey is created, you need the `provider` and `signer` props that the Safe\{Core\} SDK require to instantiate the kits.

Reviewer: ```suggestion
  Once the passkey is created, you need the `provider` and `signer` properties required to instantiate the Safe\{Core\} SDK kits.
```

  [How to initialize the Protocol Kit](https://docs.safe.global/reference-sdk-protocol-kit/initialization/init)

  The `provider` (eip1193 compatible or rpc) can be obtained by using `viem` in several ways:
  {/* <!-- vale off --> */}

  <Tabs items={['eip1193', 'rpcUrl']}>
    <Tabs.Tab>
    ```typescript
    import { createWalletClient, http } from 'viem'
    import { sepolia } from 'viem/chains'

    const walletClient = createWalletClient({
      chain: sepolia,
      transport: http('https://rpc.ankr.com/eth_sepolia')
    })

    const provider = { request: walletClient.request }
    const signer = passkeySigner
      ```
          </Tabs.Tab>
    <Tabs.Tab>
    ```typescript
    import { sepolia } from 'viem/chains'

    const provider = sepolia.rpcUrls.default.http[0]
    const signer = passkeySigner

>     ```

Reviewer: I think this is out of scope for this PR.
The used `provider` is just an example to provide to the kits, all the different ways to create it are covered in the reference here: https://docs.safe.global/reference-sdk-protocol-kit/initialization/init#provider

In fact, I think the `provider` could also be reemoved as this is just about the generating the `signer`. This also applies to all the Safe Signers guides, so I think it's out of scope for this PR.

Author: What is ooo? the two examples or the next section "Instantiate the SDK"?

Reviewer: The changes inside ```<Tabs items={['eip1193', 'rpcUrl']}>```

Author: done

    </Tabs.Tab>
  </Tabs>

  {/* <!-- vale on --> */}

  ### Instantiate SDK

  With the `provider` and `signer` you are ready to instantiate any of the kits from the Safe\{Core\} SDK and set up or use this signer as a Safe owner.

  For example, you can instantiate the `protocol-kit` as follows and sign a transaction with the passkey signer:

  {/* <!-- vale off --> */}

  ```typescript
  const protocolKit = await Safe.init({ provider, signer, safeAddress })
  
  const transaction = { to: '0x1234', value: '0x0', data: '0x' }
  const safeTransaction = await protocolKit.createTransaction({ transactions: [transaction] })
  const signedSafeTransaction = await protocolKit.signTransaction(safeTransaction)
  ```
  
  {/* <!-- vale on --> */}
  

>   </Steps>

Reviewer: ```suggestion
</Steps>
```
